shell/common/gin_converters/std_converter.h

```diff
@@ -182,6 +183,49 @@ struct Converter<std::map<K, V>> {
   }
 };
 
+template <typename K, typename V>
+struct Converter<std::vector<std::pair<K, V>>> {
+  static bool FromV8(v8::Isolate* isolate,
+                     v8::Local<v8::Value> value,
+                     std::vector<std::pair<K, V>>* out) {
+    if (!value->IsObject())
+      return false;
+    out->clear();
+    v8::Local<v8::Context> context = isolate->GetCurrentContext();
+    v8::Local<v8::Object> obj = value.As<v8::Object>();
+    v8::Local<v8::Array> keys = obj->GetPropertyNames(context).ToLocalChecked();
+    for (uint32_t i = 0; i < keys->Length(); ++i) {
+      v8::MaybeLocal<v8::Value> maybe_v8key = keys->Get(context, i);
+      if (maybe_v8key.IsEmpty())
+        return false;
+      v8::Local<v8::Value> v8key = maybe_v8key.ToLocalChecked();
+      v8::MaybeLocal<v8::Value> maybe_v8value = obj->Get(context, v8key);
+      if (maybe_v8value.IsEmpty())
+        return false;
+      K key;
+      V value;
+      if (!ConvertFromV8(isolate, v8key, &key) ||
+          !ConvertFromV8(isolate, maybe_v8value.ToLocalChecked(), &value))
+        return false;
+      (*out).emplace_back(std::move(key), std::move(value));
+    }
+    return true;
+  }
+
+  static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
+                                   const std::vector<std::pair<K, V>>& dict) {
+    v8::Local<v8::Object> obj = v8::Object::New(isolate);
+    v8::Local<v8::Context> context = isolate->GetCurrentContext();
+    for (const auto& it : dict) {
+      if (obj->Set(context, ConvertToV8(isolate, it.first),
+                   ConvertToV8(isolate, it.second))
+              .IsNothing())
+        break;
+    }
+    return obj;
+  }
```

Comment:
Is this used anywhere?

Reply:
No. I will remove it if you’ ok with it.

Reply:
Yep, let's leave this out unless we have a use for it.

Reply:
Build failed if I removed the `ToV8` method. Any suggestion?
```c++
In file included from ../../electron/shell/common/gin_converters/file_dialog_converter.cc:5:
In file included from ../../electron/shell/common/gin_converters/file_dialog_converter.h:8:
../../gin/converter.h:247:24: error: no member named 'ToV8' in 'gin::Converter<std::__1::vector<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>>>, void>'
  return Converter<T>::ToV8(isolate, input);
         ~~~~~~~~~~~~~~^
../../gin/converter.h:263:13: note: in instantiation of function template specialization 'gin::ConvertToV8<std::__1::vector<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>>>>' requested here
  *output = ConvertToV8(isolate, input);
            ^
../../gin/dictionary.h:47:10: note: in instantiation of function template specialization 'gin::TryConvertToV8<std::__1::vector<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>>>>' requested here
    if (!TryConvertToV8(isolate_, val, &v8_value))
         ^
../../electron/shell/common/gin_converters/file_dialog_converter.cc:70:8: note: in instantiation of function template specialization 'gin::Dictionary::Set<std::__1::vector<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>>>>>>' requested here
  dict.Set("filters", in.filters);
       ^
1 error generated.
```

Reply:
I have to add it back since I got build failures with it.

Reply:
It seem this method is conflicting with the filters converter from file_dialog_converter.

Reply:
Fixed now.